### PR TITLE
Incremental copy

### PR DESCRIPTION
**Pull request by mponomar**
Copy only the pages that changed, when copying on top of an existing image.

### CODE REVIEW

File: tools/comdb2ar/comdb2ar.cpp [hunk 1]
```diff
@@ -236,6 +241,14 @@ int main(int argc, char *argv[])
                 mode = CREATE_MODE;
                 break;
 
+            case 'p':
+                mode = PARTIAL_CREATE_MODE;
+                break;
+
+            case 'P':
```
Comment (nirbhayc) [inline tools/comdb2ar/comdb2ar.cpp]: p & P could be confusing. I'd go with 'i' for incremental create and 'r' restore.

Comment (mponomar) [reply tools/comdb2ar/comdb2ar.cpp]: We already have -I.  I was trying to avoid confusion there.

File: db/copycomdb2 [hunk 2]
```diff
@@ -139,14 +141,18 @@ if [[ ! -x $comdb2ar ]]; then
    comdb2ar=$(which comdb2ar 2>/dev/null)
 fi
 
+if [[ -z "$comdb2ar" ]]; then
```
Comment (mponomar) [inline db/copycomdb2]: I sure hope this fails review :)

File: tools/comdb2ar/comdb2ar.cpp (hunk 1 above)
Comment (mponomar) [reply tools/comdb2ar/comdb2ar.cpp]: FWIW, I don't expect comdb2ar to be used outside of copycomdb2

**Comment by markhannum** [comment]
This is a really fantastic idea .. it'll allow us to reduce the size of a copy to the current logical differences between the source and the destination.  Wow!

**Pull request closed without merging**